### ISSUE TITLE
[ErrorException] Undefined property: LEtudiant\Composer\Installer\SharedPackageInstaller::$binaryInstaller

### ISSUE THREAD

**Issue by mookman288** [Letudiant/composer-shared-package-plugin#13]
Hi, 

I should preface that I'm trying to install a proprietary system from my work. I'll try to provide as much information as possible without divulging their code. 

```
 [ErrorException] 
 Undefined property: LEtudiant\Composer\Installer\SharedPackageInstaller::$binaryInstaller
```

I'm currently running Windows 7 x64, PHP 7.0.2. This error originally appeared on an older build of Composer, but had worked in the past (as early as last week, same PHP build, same composer build.) In an attempt to fix the solution I entered `composer install --no-custom-installers` which allowed the system to continue installation, but the underlying application could not continue. 

I'm not entirely sure what this error means, could you expand on it? 

After upgrading composer to the latest, I'm getting the following error:

```
 [ErrorException]
 symlink(): Could not fetch file information(error 3)
```

I'm not entirely sure what this error means either, but I get the feeling it's PHP or composer related, and not necessarily related to your library. Have you encountered this before? 

I'm able to get past this part of the installation by entering `composer install --no-plugins` which allows me to continue. 

Any ideas what might be going on?

**Comment by Divi** [comment]
Hello, the first error is a BC change in the last Composer version. Just update to the last Composer version like you done. Otherwise, require the last `2.x` version of this plugin.

For the symlink error, it appears your folder or symlink is not readable or the path is wrong.
Is your dependencies folder exists ?

**Comment by mookman288** [comment]
Hi Divi, 

The dependency folder exists. If I run the application with `--no-plugins` flag, then it downloads and puts everything in the correct shared folder. 

The application does run, but I'm worried that whatever should run during the installer script is not.

**Comment by mookman288** [comment]
Divi, 

Is this a windows symlink problem? It appears to work fine on unix environments.

**Comment by Divi** [comment]
Yep, symlinks are pretty broken on Windows :(

**Comment by mookman288** [comment]
The symlink appears to be needed, but I'm not sure how to make that symlink work properly to satisfy this plugin without it throwing an error. 

If the reason why this error is being displayed (`symlink(): Could not fetch file information(error 3)`) is because I'm using Windows, then my recommendation would to be add the Windows incompatibility to the documentation to let others know of the same problem should they encounter it. 

Thank you for all of your help.

**Comment by Divi** [comment]
No, it worked on my Windows a few months ago, I said it pretty broken because relative symlinks doesn't not work.

Try with this configuration : `"symlink-base-path": "/foo/bar/composer-dependencies"`.
See : https://github.com/Letudiant/composer-shared-package-plugin/blob/master/docs/all-available-configurations.md

Can you paste me your `composer.json` file please ?
